Proofs/Experiments/MemoryAliasing.lean

```diff
@@ -366,17 +382,37 @@ theorem mem_separate_move_of_lt_of_le  (h : mem_separate' a an b bn)
 
 end MathProperties
 
+
+
+section PairwiseSeparate
+  /- Check that a direct implication of the pairwise separation is proven. -/
+  theorem pairwise_direct (h : Memory.Region.pairwiseSeparate [⟨a, 100⟩, ⟨b, 200⟩, ⟨c, 300⟩, ⟨d, 400⟩]) :
+    mem_separate' a 100 b 200 := by
+    simp_mem
+
+  /- Check that a direct implication of the pairwise separation is proven. -/
+  theorem pairwise_subset (h : Memory.Region.pairwiseSeparate [⟨a, 100⟩, ⟨b, 200⟩, ⟨c, 300⟩, ⟨d, 400⟩]) :
+    mem_separate' a 80 b 100 := by
+    simp_mem
+
+end PairwiseSeparate
+
 namespace MemOptions
 
+set_option trace.simp_mem true
+set_option trace.simp_mem.info true
```

Comment:
```suggestion
-- set_option trace.simp_mem true
-- set_option trace.simp_mem.info true
```

Reply:
Oh wait, do you need these to be turned on for the `#guard_msgs` below?

Reply:
Still, I scoped it with `set_option trace.simp_mem true in`...